tapir/shifts/templates/shifts/shift_filters.html

```diff
@@ -16,13 +16,18 @@ <h5 class="card-header">{% translate "Filters" %}</h5>
                    value="needs-help" id="btn_most_help">
             <label class="btn btn-outline-dark" for="btn_most_help">{% translate 'Most help needed' %}</label>
         </div>
-        </p>
+        </fieldset>
+        <!-- If 'Has a free slot' -->
         <p>
+
+        </p>
         <div class="btn-group-vertical" data-bs-toggle="buttons" id="slot-name-group">
-            <input type="radio" class="btn-check active-on-load" name="slot_name_options" autocomplete="off"
+        <fieldset>
```

Comment:
You need to move `class="btn-group-vertical" data-bs-toggle="buttons" id="slot-name-group"` from the containing `div` to the `fieldset` to keep the link between the buttons.